src/BuildSqlGroupByClauseCapableTrait.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Sql;
+
+use Dhii\Exception\InternalExceptionInterface;
+use Dhii\Storage\Resource\Sql\EntityFieldInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+use OutOfBoundsException;
+use OutOfRangeException;
+use stdClass;
+use Traversable;
+
+/**
+ * Common functionality for building GROUP BY query portions.
+ *
+ * @since [*next-version*]
+ */
+trait BuildSqlGroupByClauseCapableTrait
+{
+    /**
+     * Builds the GROUP BY portion of the an SQL query.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $grouping A list of strings, stringable objects or entity-field instances.
+     *
+     * @return string The built GROUP BY query portion.
+     *
+     * @throws InvalidArgumentException   If the argument is not a valid iterable.
+     * @throws OutOfRangeException        If an element in the iterable is invalid.
+     * @throws InternalExceptionInterface If a problem occurred while trying to retrieve a column name.
+     */
+    protected function _buildSqlGroupByClause($grouping = [])
+    {
+        $grouping = $this->_normalizeIterable($grouping);
+        $parts    = [];
+
+        foreach ($grouping as $_groupField) {
+            // Get entity (if any) and field
+            $_entity = ($_groupField instanceof EntityFieldInterface)
+                ? $_groupField->getEntity()
+                : null;
+            $_field  = ($_groupField instanceof EntityFieldInterface)
+                ? $_groupField->getField()
+                : $_groupField;
+
+            try {
+                // Change field to column
+                $_column = $this->_getSqlColumnName($_field);
+                // Generate SQL escaped entity field string
+                $_escaped = $this->_escapeSqlReference($_column, $_entity);
```

Comment:
Isn't it also necessary to get the table that is mapped to the entity?

Reply:
No

Reply:
Apparently, all mapping of this kind is done through table aliases.